src/extensions/projectConfig.ts

```diff
@@ -29,6 +31,43 @@ module.exports = (toolbox: CycliToolbox) => {
     return undefined
   }
 
+  const nodeVersionFileInDirectory = (dir: string): string | undefined => {
+    if (filesystem.exists(join(dir, '.nvmrc'))) {
+      return join(dir, '.nvmrc')
+    } else if (filesystem.exists(join('dir', '.node-version'))) {
+      return join(dir, '.node-version')
+    } else {
+      const packageJson = filesystem.read(join(dir, 'package.json'), 'json')
+      if (packageJson?.volta?.node || packageJson?.engines?.node) {
+        return join(dir, 'package.json')
+      }
+    }
+    return undefined
+  }
+
+  const nodeVersionFile = (context: ProjectContext): string => {
+    const { repoRoot, packageRoot } = context.path
+
+    // First, we look for node version in package root.
+    // If not found, we look for it in repository root (for monorepo support).
+    // If still not found, create .nvmrc in package root with default node version (v20.17.0).
+    const file =
+      nodeVersionFileInDirectory(packageRoot) ??
+      nodeVersionFileInDirectory(repoRoot)
+
+    if (!file) {
+      filesystem.write(join(packageRoot, '.nvmrc'), DEFAULT_NODE_VERSION)
+
+      toolbox.interactive.warning(
+        'No node version file found. Created .nvmrc with default node version (v20.17.0).'
```

Comment:
pls change hardcode for the variable + I'm thinking about adding this to what's next too - I think setting node version might be quiet a big change

Reply:
I agree, changed hardcoded variable to constant and added info to further actions box in 7f56f1810e19ed928420572080d4411263f36086